### ISSUE TITLE
E2015: Empty string default for CommaDelimitedList should not raise an error based on AllowedPattern

### ISSUE THREAD

**Issue by gautaz** [aws-cloudformation/cfn-lint#4396]
### CloudFormation Lint Version

cfn-lint 1.46.0

### What operating system are you using?

Linux hepao 6.18.16 #1-NixOS SMP PREEMPT_DYNAMIC Wed Mar  4 12:25:12 UTC 2026 x86_64 GNU/Linux

### Describe the bug

The following parameter declaration:
```yaml
  AuthorizedSourceIps:
    AllowedPattern: '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+/[0-9]+'
    Default: ''
    Type: CommaDelimitedList
```

Fails with:
```terminal
E2015 Default should be allowed by AllowedPattern
```

From my understanding, the default is here an empty list so there is no element to match against the AllowedPattern.
Why is E2015 raised?

### Expected behavior

I would expect the default empty string to be OK with the AllowedPattern for a CommaDelimitedList.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Parameters:
  AuthorizedSourceIps:
    AllowedPattern: '[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+/[0-9]+'
    Default: ''
    Type: CommaDelimitedList
```

**Comment by gautaz** [comment]
@kddejong Thanks for this later test and sorry for the hassle.
I was interpreting the same exact sentence as "there is no value to match against the pattern so it should pass".
This is a rather strange choice regarding comma delimited lists as it implies that there is no way to produce an empty list (which seems counterintuitive).
